shared/src/temp_path.rs

```diff
@@ -0,0 +1,36 @@
+use std::path::PathBuf;
+
+use rand::{distributions::Alphanumeric, Rng};
+
+const FILE_PREFIX: &str = "yazi-";
+const ID_LEN: usize = 10;
+
+/// generate a path under the system temporary directory
+///
+/// `ext`: an ASCII string
+pub fn temp_path(ext: Option<&str>) -> PathBuf {
+	let (ext, suffix_len) = ext.map(|ext| (ext, ext.len())).unwrap_or_default();
+
+	let mut name = String::with_capacity(FILE_PREFIX.len() + ID_LEN + suffix_len);
+	name.push_str(FILE_PREFIX);
+	rand::thread_rng().sample_iter(&Alphanumeric).take(ID_LEN).for_each(|c| name.push(c as char));
+
+	if !ext.is_empty() {
+		name.push('.');
+		name.push_str(ext);
+	}
+
+	let tmp = std::env::temp_dir();
```

Comment:
Instead of `temp_dir`, you can use `BOOT.cache_dir`, which is automatically created at booting.